frontend/src/components/RepositorySelectionPanel.vue

```diff
@@ -36,81 +36,82 @@
 </template>
 
 <script lang="ts">
+export default { name: "RepositorySelectionPanel" };
+</script>
+
+<script setup lang="ts">
 import { useStore } from "vuex";
-import { reactive, computed, PropType, watchEffect } from "vue";
-import { ExternalRepositoryInfo, ProjectRepositoryConfig } from "../types";
+import { reactive, computed, watchEffect } from "vue";
+import {
+  ExternalRepositoryInfo,
+  OAuthToken,
+  ProjectRepositoryConfig,
+} from "../types";
 
 interface LocalState {
   repositoryList: ExternalRepositoryInfo[];
   searchText: string;
 }
 
-export default {
-  name: "RepositorySelectionPanel",
-  props: {
-    config: {
-      required: true,
-      type: Object as PropType<ProjectRepositoryConfig>,
-    },
-  },
-  emits: ["next"],
-  setup(props, { emit }) {
-    const store = useStore();
-    const state = reactive<LocalState>({
-      repositoryList: [],
-      searchText: "",
-    });
+const props = defineProps<{ config: ProjectRepositoryConfig }>();
+
+const emit = defineEmits<{
+  (event: "next"): void;
+  (event: "set-token", payload: OAuthToken): void;
+  (event: "set-repository", payload: ExternalRepositoryInfo): void;
+}>();
+
+const store = useStore();
+const state = reactive<LocalState>({
+  repositoryList: [],
+  searchText: "",
+});
 
-    const prepareRepositoryList = () => {
-      if (props.config.vcs.type == "GITLAB_SELF_HOST") {
+const prepareRepositoryList = () => {
+  if (props.config.vcs.type == "GITLAB_SELF_HOST") {
+    store
+      .dispatch("auth/exchangeOAuthToken", {
+        vcsId: props.config.vcs.id,
+        code: props.config.code,
+      })
+      .then((token: OAuthToken) => {
+        emit("set-token", token);
         store
           .dispatch("gitlab/fetchProjectList", {
             vcs: props.config.vcs,
-            token: props.config.token.accessToken,
+            token: props.config.token,
```

Comment:
if we are not mutating props.config.token directly. Maybe just use "token: token" instead of "token: props.config.token"